src/config/midi.rs

```diff
@@ -1,19 +1,69 @@
-use crate::Result;
-use serde::Deserialize;
-use wmidi::Channel;
+use serde::{Deserialize, Deserializer};
+use std::{convert::TryInto, ops::Deref};
+use wmidi::U7;
 
 #[derive(Debug, Deserialize)]
 pub struct Midi {
-    pub port: String,
-    channel: u8,
+    pub port: Option<String>,
 }
 
 impl Midi {
-    pub fn default_channel() -> Channel {
-        Channel::Ch1
+    pub fn default() -> Self {
+        Self { port: None }
     }
+}
+
+#[derive(Copy, Clone, Debug, Deserialize)]
+pub struct MidiSlider {
+    pub channel: Channel,
+    pub control_change: ControlChange,
+}
+
+#[derive(Copy, Clone, Debug, Deserialize)]
```

Comment:
I wouldn't worry too much about the fancy stuff going on here with `Channel` and `ControlChange`. The summary is, I wanted to be able to deserialize `wmidi`'s types for those things, but you can't implement traits for types you didn't create (so I can't implement`Deserialize` for them), so I had to create wrapper types around them which delegate deserialization of those types out to custom functions.